packages/flutter/lib/src/material/snack_bar.dart

```diff
@@ -480,10 +480,9 @@ class _SnackBarState extends State<SnackBar> {
         onError: colorScheme.error,
         brightness: brightness,
       ),
-      snackBarTheme: snackBarTheme,
     );
 
-    final TextStyle? contentTextStyle = snackBarTheme.contentTextStyle ?? inverseTheme.textTheme.subtitle1;
+    final TextStyle? contentTextStyle = snackBarTheme.contentTextStyle ?? ThemeData(brightness: brightness,).textTheme.subtitle1;
```

Comment:
Why isn't this `theme.textTheme.subtitle1`?  If this version produces a text style with a different color (I can't see that, but I probably haven't looked closely enough), then why not compute the foreground color and use `theme.textTheme.subtitle1.copyWith(color: foregroundColor)`?

Reply:
This version produces a test style with different color and size by default.
Additionally, If the developer provides a custom `subtitle1`, it will change `SnackBar` behavior

Reply:
Understood that the color would be different, that's why I suggested computing the foreground color etc.  Why would the size be different?  As far as I can tell, `theme.textTheme` is just the top-level theme's text theme. If the developer has overridden that, then we should honor the override. 

Note also: SnackbarTheme has a contentTextStyle. For the sake of consistency, Snackbar should too.

Reply:
`ThemeData(brightness: brightness,).textTheme.subtitle1` is:
`TextStyle(debugLabel: whiteMountainView subtitle1, inherit: true, color: Color(0xffffffff), family: Roboto, decoration: TextDecoration.none)`

`theme.textTheme.subtitle1` is (comes from `MaterialApp`):
`TextStyle(debugLabel: (englishLike subhead 2014).merge(blackMountainView subtitle1), inherit: false, color: Color(0xdd000000), family: Roboto, size: 16.0, weight: 400, baseline: alphabetic, decoration: TextDecoration.none)`

And if we change this style to `theme.textTheme.subtitle1`, will cause 18 test cases failed:
![image](https://user-images.githubusercontent.com/61075224/99762718-170db480-2b34-11eb-8179-355e4213d28c.png)

Reply:
Any thoughts about this, Keep as is or breaking it?

Reply:
I tried to dive into why the values differ and couldn't quite figure it out, although the value from `theme.textTheme.subtitle` [looks more correct](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/typography.dart#L507). The only correct thing about `ThemeData(brightness: brightness,).textTheme.subtitle1` is the color value.

Let's keep it as-is here. I think this particular change can be its own separate PR where we address the breaking changes (and migration guide), since those breaking changes will likely result in Google testing failures and the test failures in your above screenshot.

Reply:
OK, sounds good!